### ISSUE TITLE
Simplicity vs. complexity

### ISSUE THREAD

**Issue by snowshed** [Microsoft/winfile#168]
To all readers...

If I am posting this in the wrong area, no offense intended.  I'm not a programmer, I'm just a retired "senior" who would like to offer some comments on this project.  I am a "user", I like to see what I can make the computer do for me.  I bought my first computer in 1983.  <G>

I just learned of this program @3 days ago.  You have no idea how pleased I was to discover it existed.

I do computer tutoring on a part-time basis.  Usually to seniors and others that have so little computer knowledge it's dismaying.

I've found, that when trying to teach hard drive organization and file management using the current versions of Windows/File Explorer is very confusing to people who are basically in kindergarten.

The very simplicity of File Manager is, or could be, the best alternative file manager program out there to make explaining organization and management that I think exists.

I do not know how this program actually works, operates, looks, etc., but for beginners, having Libraries , personal folders, the Network, etc. displaying in the Navigation pane simply is confusing to those with little or no knowledge.

So, I would like to encourage those involved with the development of Winfile10 to keep the UI as simple as possible.  My... disgust, I guess... is the seeming assumption there are no beginners to computers.  Which is simply not true.  So, people new to computers end up trying to learn starting at the junior high level, when they are actually just in kindergarten.

Regarding my comment about not knowing how the program works...  I can't the program to run!  :-(  Windows 8.1 fully updated.  After installing the program, and the Visual Studio .dll, I got the message the dll could not be found.  Searching the web, I learned the dll needed to be in the SysWOW64(?) folder.  Copied it there, then got the message the program could not be started.  Error Code 0x000007b, if I remember correctly.  More searching, and the suggestion to run sfc/scannow.  Yep, corrupted system files.  I don't know how to read and understand the log file, so I just refreshed 8.1.  No more corrupted files, but still got the same error code.

To date, none of the suggested solutions to the error code I've found have failed to solve the problem.

If some one has a suggestion to get the program to run for me, it would be very much appreciated.

Ken

**Comment by snowshed** [comment]
64 bit OS.

Installed the 64 bit redistributable, I gather I should have installed the 32 bit version?

If that's true, do I need to remove the 64 bit version before installing the 32 bit version?  The redistributable does not show up in Programs and Features.

**Comment by ZanderBrown** [comment]
No you can have the 32 & 64 bit versions installed side-by-side

**Comment by snowshed** [comment]
Hi, Zander,

Tried installing the 32 bit version, got a message I couldn't install it because another version of the program was running.

Removed the 64 bit, didn't make any difference.

The link from Winfile installs 14.0.23026.  However, something installed 14.0.24212.  Could the 24212 package be the cause of the error message?

**Comment by clzls** [comment]
Tip: Do not use @ symbol unless you are referring someone. Abusing it may disturb someone else (for example the user who named "3" in this example.)

> I just learned of this program @ 3 days ago.

And, if you are not a professional computer engineer, DO NOT TOUCH DLLs DIRECTLY. Use tools provided by Microsoft or trusted software companies.
(VC Redistributables are totally a mess, I think. Even professionals may mistreat them.)

**Comment by snowshed** [comment]
HI, Cizis,

Anyone who gets disturbed at how the "@" should do some research.  :-)  As I used it, the symbol is interpreted to represent the word "about", and has been for years, long before computer and social media use.  So my sentence should be read as "I just learned of the program about 3 days ago."  It's also an accounting symbol.

There's an example of the sign being used in 1448 in this Wikipedia article...  https://en.wikipedia.org/wiki/At_sign

I'm not doing anything "directly" to any DLL.  This assumes we have the same understanding of the word as you used it.

**Comment by clzls** [comment]
I'm a Chinese so my words may have misled you. When you use the symbol @ , the GitHub system will automatically send an email to the certain user, in this example, the user "3". He will get a email containing all your words above.

获取 Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by ad43** [comment]
I agree with Ken that Library and Photos (etc) artifacts should not be visible in Winfile (by default, anyway), because surely it's a File Manager not a chapter index.

We once had a Program Manager to allow us to construct groups of links to certain sets of information or projects or business workflow and relevant applications. These were uncluttered with superfluous menus, navigation panes and status lines. The icons would stay stable in one layout too, not get moved around unexpectedly as in Windows 10's shell (an absolute pain when setting up non-technical business user's desktop and start menu layouts, especially with Roaming Profiles).

Explorer allows folders to be presented on the desktop containing links, but the clutter and layout instability makes them unappealing.

Now there's a thought - anyone got the time or enthusiasm to resurrect Program Manager, if Microsoft has no further interest in it? Example screen:
   http://chorusofone.no-ip.org/Computerstuff/Images/progman10.PNG

Chris

**Comment by snowshed** [comment]
Hi, cizis,

I did not know GitHub operated that way.  IMO, it's a bug when a program "assumes" certain characters or words mean only one thing, and program accordingly.

GitHub may want to review how they handle the "at" sign, as I'm not going to change 50 years of using the sign in one of its many uses just so GitHub doesn't sent something out that it shouldn't.  <G>

**Comment by snowshed** [comment]
Hi, ad43,

Ah, Program Manager.  I'd forgotten about that part of Windows...  And, never thought much of it either.  LOL

Put all of those "extra" folders that create the clutter on your desktop in a "master/top level" folder, and simply minimize the folder when you want the clutter out of the way.  Wouldn't that accomplish the essence of Program Manager?

**Comment by clzls** [comment]
You can prevent that by just adding a space(" ") after @ . It's not a bug; it's a feature. In fact, we are not writing plain text, we are writing in GitHub-flavored Markdown language. We must follow the **grammer** of it.
________________________________

**Comment by ZanderBrown** [comment]
@ad43 progman was brought up previously, unlikely we will get original source and the file format was problematic IIRC. That said I would like to see a reimplementation (WinFile should provide a good example of Win32 MDI)

@snowshed Yes & No, `progman` used a new window for each group within the MDI host. Quite different from multiple explorer windows (also didn't support nesting groups (folders) and had no trees or toolbars)

**Comment by snowshed** [comment]
Hi, clzls,

My apologies for the mistake on your name.  I have macular degeneration in both eyes.  With everyone's fascination of using a sans-serif font, a lower case "i" and "l", plus the number "1", depending on the particular sans-serif font, is often indistinguishable to me.

I will never put a space between the "at" sign and the following character unless that would represent correct written English.  GitHub is simply not that important to me.  <G>

But getting Winfile10 to run for me is, which hasn't been accomplished as of yet.  :-)

**Comment by clzls** [comment]
@ZanderBrown , I think compiling it by MinGW will get rid of all these disgusting VC Redistributable. I don't think Winfile need any advanced feature of VC++.
@snowshed , This usage of @ is common on the Internet now. But never mind, you can just use it in your own way.

As a reference, here's all the VC Redistributables on my computer. (Quite annoying, isn't it...)
![image](https://user-images.githubusercontent.com/16474386/39952812-844f8234-55d1-11e8-9530-4ef726642a24.png)

**Comment by zhw2101024** [comment]
A statically compiled version binary may fix the dll hell,since WinFile only use win32 apis.
